risc0/core/src/field/mod.rs

```diff
@@ -122,6 +128,12 @@ pub trait Elem:
         self
     }
 
+    /// Returns this element, but checks to make sure it's in reduced form.
+    fn ensure_reduced(&self) -> &Self {
+        assert!(self.is_reduced());
```

Comment:
I wonder if we should use `debug_assert!`?

Reply:
It would depend on how how this is used, but I would air on the side of having it be a full assert. However, now that I look, I don't see this function used anywhere, so I would actually suggest we simply remove it.

Reply:
Ah, good call, this is remnants of an earlier approach. Removing this also reduces potential confusion between `ensure_reduced` (`assert` if >= prime) and `ensure_valid` (`debug_assert` if `0xffffffff`).